library/std/src/sys/windows/thread_local_key.rs

```diff
@@ -111,16 +111,13 @@ struct Node {
 }
 
 unsafe fn register_dtor(key: Key, dtor: Dtor) {
-    let mut node = Box::new(Node { key, dtor, next: ptr::null_mut() });
+    let mut node = ManuallyDrop::new(Box::new(Node { key, dtor, next: ptr::null_mut() }));
 
     let mut head = DTORS.load(SeqCst);
     loop {
         node.next = head;
-        match DTORS.compare_exchange(head, &mut *node, SeqCst, SeqCst) {
-            Ok(_) => {
-                mem::forget(node);
-                return;
-            }
+        match DTORS.compare_exchange(head, &mut **node, SeqCst, SeqCst) {
```

Comment:
This isn't really related, but shouldn't we use `compare_exchange_weak` here ?

Reply:
Possible; I don't really have a clear idea for what should be used when.
Since it's not related, that should then probably be fixed by someone else in a separate PR. :)

Reply:
Fine, I'll do it when this PR is merged.

Reply:
@a1phyr Did this end up happening?